packages/amplify-provider-awscloudformation/src/utils/ssm-utils/env-parameter-ssm-helpers.ts

```diff
@@ -0,0 +1,114 @@
+import { $TSContext, AmplifyFault, stateManager } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import type { SSM as SSMType } from 'aws-sdk';
+import { SSM } from '../../aws-utils/aws-ssm';
+import { resolveAppId } from '../resolve-appId';
+import { executeSdkPromisesWithExponentialBackoff } from './exp-backoff-executor';
+
+/**
+ * Higher order function for uploading CloudFormation parameters to the service
+ */
+export const getEnvParametersUploadHandler = async (
+  context: $TSContext,
+): Promise<((key: string, value: string | boolean | number) => Promise<void>) | undefined> => {
+  let appId: string;
+  try {
+    appId = resolveAppId(context);
+  } catch {
+    printer.warn('Failed to resolve AppId, skipping parameter upload.');
+    return undefined;
```

Comment:
Would it eliminate some upstream defensive logic if this function just returned a noop function instead of undefined if no appId is found? (and same with the download handler)

Reply:
It doesn't eliminate much, but it simplifies the return type so I went with your suggestion